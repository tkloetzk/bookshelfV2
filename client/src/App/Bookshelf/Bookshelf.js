import React, { useEffect } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import Results from '../Results/Results'
import { getBookshelf } from '../../store/bookshelf/bookshelfActions'

export default function Bookshelf() {
  const bookshelf = useSelector(state => state.bookshelf.bookshelf)
  const dispatch = useDispatch()

  useEffect(() => {
    dispatch(getBookshelf())
  }, [dispatch])

  return <Results booklist={bookshelf} />
}
