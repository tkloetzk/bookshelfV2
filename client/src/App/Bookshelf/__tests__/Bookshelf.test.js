import { MuiThemeProvider } from '@material-ui/core/styles'
import React from 'react'
import { render, cleanup } from '@testing-library/react'
import configureMockStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import muiTheme from '../../../config/themeConfig'
import Bookshelf from '../Bookshelf'
import '@testing-library/jest-dom/extend-expect'

const mockStore = configureMockStore()

describe('Bookshelf', () => {
  let book
  beforeEach(() => {
    book = {
      amazonAverageRating: 4.5,
      amazonRatingsCount: 553,
      categories: ['Family & Relationships'],
      description:
        'Answers the most important parenting questions about raising children from birth through kindergarten, covering such topics as discipline, sleeping, day care, safety, independence, and feeding.',
      differences: [
        {
          currentValue: 3.9,
          key: 'amazonAverageRating',
          newValue: 4.5,
        },
      ],
      goodreadsAverageRating: 4,
      goodreadsRatingsCount: 40,
      isbn: '9781402218279',
      price: '',
      subtitle:
        'From Birth to Kindergarten, Answers to the Top 150 Questions about Raising a Young Child',
      thumbnail:
        'http://books.google.com/books/content?id=oD0omQEACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api',
      title: 'The New Baby Answer Book',
    }
  })

  afterEach(cleanup)

  describe('render', () => {
    it('should render as expected if bookshelf is empty', () => {
      const store = mockStore({
        bookshelf: { bookshelf: [] },
      })
      store.dispatch = jest.fn()

      const { asFragment } = render(
        <Provider store={store}>
          <MuiThemeProvider theme={muiTheme}>
            <Bookshelf />
          </MuiThemeProvider>
        </Provider>
      )
      expect(store.dispatch).toHaveBeenCalledTimes(1)
      expect(asFragment()).toMatchSnapshot()
    })
    it('should render as expected if bookshelf has books', () => {
      const store = mockStore({
        bookshelf: {
          bookshelf: [book],
        },
      })
      store.dispatch = jest.fn()
      const { asFragment } = render(
        <Provider store={store}>
          <MuiThemeProvider theme={muiTheme}>
            <Bookshelf />
          </MuiThemeProvider>
        </Provider>
      )
      expect(store.dispatch).toHaveBeenCalledTimes(1)
      expect(asFragment()).toMatchSnapshot()
    })
  })
  // describe('handleSave', () => {
  //   it('calls updateBookOnBookshelfService with correct parameters', async () => {
  //     const store = mockStore({
  //       bookshelf: { bookshelf: [book] },
  //     })
  //     store.dispatch = jest.fn()

  //     const wrapper = shallow(
  //       <Provider store={store}>
  //         <Bookshelf />
  //       </Provider>
  //     )
  //     const instance = wrapper.instance()
  //     await instance.handleSave({}, [])
  //     expect(store.dispatch).toHaveBeenCalledTimes(1)
  //   })
  // })
})
