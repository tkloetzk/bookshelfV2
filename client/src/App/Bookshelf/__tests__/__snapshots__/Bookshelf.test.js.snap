// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bookshelf render should render as expected if bookshelf has books 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root makeStyles-container-107 MuiGrid-container MuiGrid-spacing-xs-4 MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
    >
      <div
        aria-label="OWNED/READ genres"
        class="MuiToggleButtonGroup-root"
        role="group"
      >
        <button
          aria-label="OWNED"
          aria-pressed="false"
          class="MuiButtonBase-root MuiToggleButton-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedSizeSmall MuiToggleButton-sizeSmall"
          tabindex="0"
          type="button"
          value="owned"
        >
          <span
            class="MuiToggleButton-label"
          >
            OWNED
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          aria-label="READ"
          aria-pressed="false"
          class="MuiButtonBase-root MuiToggleButton-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedSizeSmall MuiToggleButton-sizeSmall"
          tabindex="0"
          type="button"
          value="read"
        >
          <span
            class="MuiToggleButton-label"
          >
            READ
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
    >
      <div
        class=" css-2b097c-container"
      >
        <div
          class=" css-yk16xz-control"
        >
          <div
            class=" css-1hwfws3"
          >
            <div
              class=" css-1wa3eu0-placeholder"
            >
              Select...
            </div>
            <div
              class="css-w8afj7-Input"
            >
              <div
                class=""
                style="display: inline-block;"
              >
                <input
                  aria-autocomplete="list"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  id="react-select-2-input"
                  spellcheck="false"
                  style="box-sizing: content-box; width: 2px; border: 0px; font-size: inherit; opacity: 1; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
                <div
                  style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-size: inherit; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                />
              </div>
            </div>
          </div>
          <div
            class=" css-1g48xl4-IndicatorsContainer"
          >
            <span
              class=" css-1okebmr-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              class=" css-tlfecz-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-tj5bde-Svg"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
    >
      <div
        aria-label="AND/OR genres"
        class="MuiToggleButtonGroup-root"
        role="group"
      >
        <button
          aria-label="OR"
          aria-pressed="true"
          class="MuiButtonBase-root MuiToggleButton-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedSizeSmall Mui-selected MuiToggleButton-sizeSmall"
          tabindex="0"
          type="button"
          value="OR"
        >
          <span
            class="MuiToggleButton-label"
          >
            OR
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          aria-label="AND"
          aria-pressed="false"
          class="MuiButtonBase-root MuiToggleButton-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedSizeSmall MuiToggleButton-sizeSmall"
          tabindex="0"
          type="button"
          value="AND"
        >
          <span
            class="MuiToggleButton-label"
          >
            AND
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation1 MuiCard-root makeStyles-card-236  makeStyles-different-237 MuiPaper-rounded"
    >
      <div
        class="makeStyles-header-238 makeStyles-headerUnread-240"
        data-testid="header"
      >
        <div
          class="MuiCardHeader-root makeStyles-cardHeader-243"
        >
          <div
            class="MuiCardHeader-avatar makeStyles-avatar-245"
          >
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              4.25
            </p>
          </div>
          <div
            class="MuiCardHeader-content makeStyles-headerContent-244"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter"
            >
              The New Baby Answer Book
            </p>
          </div>
          <div
            class="MuiCardHeader-action makeStyles-iconButton-242"
          >
            <svg
              aria-hidden="true"
              aria-label="Differences"
              class="MuiSvgIcon-root"
              currentitem="false"
              data-for="differencesIcon-9781402218279"
              data-tip="true"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 2H4c-1.1 0-2 .9-2 2v18l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm0 14H5.17L4 17.17V4h16v12zM11 5h2v6h-2zm0 8h2v2h-2z"
              />
            </svg>
            <div
              class="__react_component_tooltip place-top type-dark "
              data-id="tooltip"
              id="differencesIcon-9781402218279"
            >
              <span>
                amazonAverageRating 3.9 -&gt; 4.5
                <br />
              </span>
            </div>
          </div>
        </div>
        <span
          class="MuiTypography-root MuiTypography-caption MuiTypography-alignCenter"
        >
          Family & Relationships
        </span>
      </div>
      <img
        class="MuiCardMedia-root MuiCardMedia-media makeStyles-media-241"
        src="http://books.google.com/books/content?id=oD0omQEACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api"
        title="The New Baby Answer Book"
      />
      <div
        class="MuiCardContent-root"
      >
        <p
          class="MuiTypography-root makeStyles-description-246 MuiTypography-caption MuiTypography-colorTextSecondary"
        >
          Answers the most important parenting questions about raising children from birth through kindergarten, covering such topics as discipline, sleeping, day care, safety, independence, and feeding.
        </p>
      </div>
      <div
        class="MuiCardActions-root makeStyles-expandAction-248 MuiCardActions-spacing"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root makeStyles-expand-247"
          data-testid="expandButton"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Bookshelf render should render as expected if bookshelf is empty 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
  />
</DocumentFragment>
`;
